lib/src/sdk.ts

```diff
@@ -436,6 +451,127 @@ export class ExchangeSDK {
     return tx;
   }
 
+  /**
+   * Ask user to validate a fund transaction.
+   * @param {FundInfo} info - Information necessary to create a fund transaction.
+   * @return {Promise<string | void>} Promise of the hash of the sent transaction.
+   * @throws {ExchangeError}
+   */
+  async fund(info: FundInfo): Promise<string | void> {
+    this.flowType = "generic";
+    this.logger.log("*** Start Fund ***");
+
+    const {
+      fromAccountId,
+      fromAmount,
+      feeStrategy = FeeStrategyEnum.MEDIUM,
+      customFeeConfig = {},
+      orderId,
+      type = ExchangeType.CARD,
+    } = info;
+
+    const { account, currency } =
+      await this.walletAPIDecorator.retrieveUserAccount(
+        fromAccountId,
+        this.flowType
+      );
+
+    // Check enough funds
+    const initialAtomicAmount = this.convertToAtomicUnit(fromAmount, currency);
+    this.canSpendAmount(account, initialAtomicAmount);
+
+    // Step 1: Ask for deviceTransactionId
+    const deviceTransactionId = await this.exchangeModule
+      //TODO: pass in provider after updating startFund in LL
+      .startFund
+      // {provider: this.providerId}
+      ()
+      .catch((error: Error) => {
+        const err = parseError(this.flowType, error, StepError.NONCE);
+        this.logger.error(err as Error);
+        throw err;
+      });
+
+    this.logger.debug("DeviceTransactionId retrieved:", deviceTransactionId);
+
+    // Step 2: Ask for payload creation
+    const { recipientAddress, binaryPayload, signature, amount } =
+      await this.fundPayloadRequest({
+        orderId,
+        amount: fromAmount,
+        account,
+        deviceTransactionId,
+        type,
+      });
+
+    // Compare initial amount with payload amount
+    const fromAmountAtomic = this.convertToAtomicUnit(amount, currency);
+    this.verifyAmount(initialAtomicAmount, fromAmountAtomic);
+
+    this.logger.log("Payload received:", {
+      recipientAddress,
+      amount: fromAmountAtomic,
+      binaryPayload,
+      signature,
+    });
+
+    // Step 3: Send payload
+    const transaction = await this.walletAPIDecorator
+      .createTransaction(
+        {
+          recipient: recipientAddress,
+          amount: fromAmountAtomic,
```

Comment:
We could avoid this extra conversion, but totally up to you.

```suggestion
        this.verifyAmount(fromAmount, amount);

    this.logger.log("Payload received:", {
      recipientAddress,
      amount: initialAtomicAmount,
      binaryPayload,
      signature,
    });

    // Step 3: Send payload
    const transaction = await this.walletAPIDecorator
      .createTransaction(
        {
          recipient: recipientAddress,
          amount: initialAtomicAmount,
```

Reply:
@sarneijim I just realised that the amount we get from the payload is not actually from the payload. We pass in the initial amount to the request in `retrievePayload`. But the only data we get from there is the return of `parseFundBackendInfo` which doesn't include the amount. Basically we're just passing this value around but it's always the initial amount so we don't need to verify it. We can also remove the second `canSpendAmount` check from the `sell` method.